ext/ITensorNetworksEinExprsExt/src/ITensorNetworksEinExprsExt.jl

```diff
@@ -0,0 +1,70 @@
+module ITensorNetworksEinExprsExt
+
+using ITensors: Index, ITensor, Algorithm
+using ITensorNetworks:
+  ITensorNetworks,
+  ITensorNetwork,
+  vertextype,
+  vertex_data,
+  externalinds,
+  contraction_sequence
+using EinExprs: EinExprs, EinExpr, einexpr, SizedEinExpr
+
+function EinExprs.einexpr(tn::ITensorNetwork; optimizer::EinExprs.Optimizer)
```

Comment:
It seems like the code for `EinExprs.einexpr` and `ITensorNetworks.contraction_sequence` are pretty similar, is there some common code logic we can split off and share between them?

Reply:
I wanted `ITensorNetworks.contraction_sequence` to call `EinExprs.einexpr` but then duplicated them because `contraction_sequence` required `tensor_map`. Now I've refactored the common code to `prepare_einexpr` function.